framework/src/media/audio/audio_manager.c

```diff
@@ -504,7 +521,38 @@ int start_audio_stream_in(void *data, uint32_t frames)
 	}
 
 	do {
-		ret = pcm_readi(pcm, data, frames);
+		if (cur_card->resample.necessary) {
+			unsigned int frames_to_read = (unsigned int)((float)frames * cur_card->resample.ratio);
```

Comment:
frames is same as get_input_frame_count, framework using that, but is it added to handle decimal value?